python/lsst/ts/hvac/csc.py

```diff
@@ -410,19 +412,17 @@ def _handle_mqtt_messages(self) -> None:
 
             if topic in self.hvac_state:
                 item_state = self.hvac_state[topic][item]
-                value = payload
-                if payload not in [
-                    b"Automatico",
-                    b"Encendido$20Manual",
-                    b"Apagado$20Manual",
-                ]:
-                    value = json.loads(payload)
-                else:
-                    value = True
-
-                item_state.recent_values.append(value)
+                if payload in [
+                    "Automatico",
+                    "Encendido$20Manual",
+                    "Apagado$20Manual",
+                ] or (isinstance(payload, str) and "AUTOMATICO" in payload):
+                    payload = True
```

Comment:
This seems weird to reassign what looks to be a string to a boolean.

Reply:
Well done spotting that. Yeah, all these cases the payload type is supposed to be a boolean. This is an open issue that needs to be addressed in the HVAC server and we're waiting for the new contract to be signed so we can start working on that.